build.gradle

```diff
@@ -47,156 +65,120 @@ project.version     = 'git describe --always --dirty'.execute().text.trim()
 // -----------------------------------------------------------------------------
 
 repositories {
-    // Use jcenter for resolving dependencies.
-    // You can declare any Maven/Ivy/file repository here.
-    jcenter()
+    // Use mavenCentral for resolving dependencies.
+    mavenCentral()
 }
 
 dependencies {
     def versions = [
-            immutablesVersion: '2.7.4',
+        immutablesVersion: '2.9.3',
     ]
 
     // This dependency is exported to consumers, that is to say found on their compile classpath.
     implementation(
             // General
-            'com.google.guava:guava:27.0.1-jre',
-    )
+            'com.google.guava:guava:23.0',
+            )
 
     compileOnly(
             "org.immutables:value:${versions.immutablesVersion}",
-    )
+            )
 
     testImplementation(
             // Use the awesome Spock testing and specification framework even with Java
-            'org.spockframework:spock-core:1.2-groovy-2.5',
-            'junit:junit:4.12',
+            'org.spockframework:spock-core:2.4-M1-groovy-4.0',
+            'junit:junit:4.13.2',
```

Comment:
Is there a reason we haven't gone to JUnit 5?
Sounds like it'll work fine with spock:
https://blog.jdriven.com/2018/10/combining-spock-junit-5-tests/

Reply:
The `useLatestVersionsCheck` gradle task didn't pick up the lastest version of JUnit as it was moved from `junit/junit` to `org.junit.jupiter/junit-jupiter-api` (The latest version of JUnit on the old repo was 4.13.2). 
Updated to pull from the new location.